sentry-android-core/src/main/java/io/sentry/android/core/internal/gestures/SentryGestureListener.java

```diff
@@ -168,6 +184,58 @@ private void addBreadcrumb(
             eventType, ViewUtils.getResourceId(target), className, additionalData));
   }
 
+  private void startTracing(final @NotNull View target, final @NotNull String eventType) {
+    if (!performanceEnabled) {
+      return;
+    }
+
+    final Activity activity = currentActivity.get();
+    if (activity == null) {
+      options.getLogger().log(SentryLevel.DEBUG, "Activity is null, no transaction captured");
+      return;
+    }
+
+    // as we allow a single UI transaction running on the bound Scope, we finish the previous one
+    if (activeTransaction != null) {
+      activeTransaction.finish();
+    }
```

Comment:
Wondering if that makes sense, I'd rather let the current one run, we can discuss that today.

Reply:
yeah, as we discussed If it's the same view -> we keep the transaction running and finish if it's a new view.